src/System.Globalization/tests/NumberFormatInfo/NumberFormatInfoData.cs

```diff
@@ -93,5 +97,22 @@ internal static int[] GetCurrencyNegativePatterns(CultureInfo cultureInfo)
 
             throw DateTimeFormatInfoData.GetCultureNotSupportedException(cultureInfo);
         }
+        
+        private static Version GetOSXKernelVersion()
+        {
+            if (s_isOSX)
+            {
+                ulong bytesLength = 256;
+                byte[] bytes = new byte[bytesLength];
+                sysctlbyname("kern.osrelease", bytes, ref bytesLength, null, 0);
+                string versionString = Encoding.UTF8.GetString(bytes);
+                return Version.Parse(versionString);
+            }
+            
+            return new Version(0, 0, 0);
+        }
+        
+        [DllImport("libc")]
+        private static extern int sysctlbyname(string ctlName, byte[] oldp, ref ulong oldpLen, byte[] newp, ulong newpLen);
```

Comment:
>  if we misunderstand the structure of the data or how it's meant to be interpreted, we're likely to make that mistake in both places.

This is possible but still current way of testing is not helping with this issue either. usually people very activate in complaining about the globalization data so if we run in such case most likely someone will report it even our test didn't catch it. the current way the test is written means any change in the OS side we have to go and fix the test again and again which I don't think this is good to do. 

> This is also just one example of a much larger issue
I agree with you we may run into other scenarios which may need checking the OS type and version but I wanted to clarify for this specific globalization issue, this is not really needed because if any dev write code depend on that will broken sooner or later.